client/state/purchases/actions.js

```diff
@@ -0,0 +1,136 @@
+// External dependencies
+import i18n from 'i18n-calypso';
+
+// Internal dependencies
+import olark from 'lib/olark';
+import {
+	PRIVACY_PROTECTION_CANCEL,
+	PRIVACY_PROTECTION_CANCEL_COMPLETED,
+	PRIVACY_PROTECTION_CANCEL_FAILED,
+	PURCHASES_REMOVE,
+	PURCHASES_SITE_FETCH,
+	PURCHASES_SITE_FETCH_COMPLETED,
+	PURCHASES_SITE_FETCH_FAILED,
+	PURCHASES_USER_FETCH,
+	PURCHASES_USER_FETCH_COMPLETED,
+	PURCHASES_USER_FETCH_FAILED,
+	PURCHASE_REMOVE,
+	PURCHASE_REMOVE_COMPLETED,
+	PURCHASE_REMOVE_FAILED
+} from 'state/action-types';
+import purchasesAssembler from 'lib/purchases/assembler';
+import wp from 'lib/wp';
+const wpcom = wp.undocumented();
+
+const PURCHASES_FETCH_ERROR_MESSAGE = i18n.translate( 'There was an error retrieving purchases.' ),
+	PURCHASE_REMOVE_ERROR_MESSAGE = i18n.translate( 'There was an error removing the purchase.' );
+
+export const cancelPrivateRegistration = purchaseId => dispatch => {
+	dispatch( {
+		type: PRIVACY_PROTECTION_CANCEL,
+		purchaseId
+	} );
+
+	return new Promise( ( resolve, reject ) => {
```

Comment:
I seems `wpcom` calls return a Promise when you don't give a callback, maybe you could use that behavior to simplify your code and simptly 
```
return wpcom.cancelPrivateRegistration( purchaseId ).then( ...
```